lib/services/project-properties-service.ts

```diff
@@ -259,5 +257,78 @@ export class ProjectPropertiesService implements IProjectPropertiesService {
 			return properties;
 		}).future<any>()();
 	}
+
+	private notifyPropertyChanged(framework: string, propertyName: string, propertyValue: any): IFuture<void> {
+		return ((): void => {
+			let projectSchema = this.$jsonSchemaValidator.tryResolveValidationSchema(framework);
+			let propData = projectSchema[propertyName];
+			if(propData && propData.onChanging) {
+				this.$injector.dynamicCall(propData.onChanging, [propertyValue]).wait();
+			}
+		}).future<void>()();
+	}
+
+	private moveCorePluginsToConfigurationSpecificData(projectData: IProjectData, configurationSpecificData:  IDictionary<IProjectData>): void {
+		if(projectData.CorePlugins && projectData.CorePlugins.length > 0) {
+			_.each(configurationSpecificData, (configurationData: IProjectData, configuration: string) => {
+				this.$logger.trace(`Move CorePlugins from project data to '${configuration}' configuration.`);
+				if(configurationData.CorePlugins && configurationData.CorePlugins.length > 0 && _.difference(configurationData.CorePlugins, projectData.CorePlugins).length !== 0) {
+					this.$errors.failWithoutHelp(`CorePlugins are defined in both '${this.$projectConstants.PROJECT_FILE}' and '.${configuration}${this.$projectConstants.PROJECT_FILE}'. Remove them from one of the files and try again.`);
+				}
+				configurationData.CorePlugins = projectData.CorePlugins;
+			});
+		}
+		delete projectData.CorePlugins;
+	}
+
+	private validateAllProjectData(projectData: IProjectData, configurationSpecificData:IDictionary<IProjectData>): void {
+		let projectConfigurations = _.keys(configurationSpecificData);
+		_.each(projectConfigurations, configuration => {
+			this.validateProjectData(projectData, configurationSpecificData[configuration]);
+		});
+
+		this.validateProjectData(projectData);
+	}
+
+	private validateProjectData(projectData: IProjectData, configurationSpecificData?: IProjectData): void {
```

Comment:
We have some validation logic in `project.ts`. Can we unify it and use only from one place?